content/2022-01-04-brigade-2.2.0.md

```diff
@@ -0,0 +1,66 @@
+---
+title: "Brigade 2.2.0"
+slug: "2022-01-04-brigade-2.2.0"
+description: "Brigade 2.2.0 has been released"
+date: "2022-01-04"
+author: "Kent Rancourt"
+summary: "Brigade 2.2.0 has been released."
+---
+
+Happy New Year, Brigadiers!
+
+We're starting the new year off right with our second minor release of Brigade 2.
+
+v2.2.0 continues the trend started with v2.1.0 by delivering just a handful of small, incremental improvements. Notably, in this release, we've corrected a bug in the user list endpoint and made it easy for script authors to designate jobs as fallible -- that is, _allowed to fail without forcing the overall workflow defined by the script to fail_. This is a great feature for cases where script authors may wish to treat a certain job's failure as informational in nature rather than fatal.
+
+To demonstrate, fallible jobs, consider this small fragment of a `brigade.ts` file:
+
+```typescript
+events.on("brigade.sh/cli", "exec", async event => {
+  let job0 = new Job("test-job0", "alpine", event)
+  job0.primaryContainer.command = ["false"]
+
+  let job1 = new Job("test-job1", "alpine", event)
+  job1.primaryContainer.command = ["true"]
+
+  await Job.sequence(job0, job1).run()
+})
+```
+
+In this example, wherein `job0` is destined to always fail, the sequence of `job0` followed by `job1` will never complete and the worker executing this script will _always_ fail. There are existing ways around this, of course, but they require some ingenuity and fair bit of code. With fallible jobs, however, this script requires only the slightest modification to ensure `job0` has no bearing on the success or failure of the sequence and the worker that's executing it:
```

Comment:
is destined to? perhaps "is configured to" or "will"? Destiny is a confusing layer to add to an already confusing matter (fallible vs destined-to-fail - apparently different?)

Reply:
"There are existing ways around this, of course, but they require some ingenuity and fair bit of code."

Why "of course"? Is there a link to info about this? What existing ways? Do I lack ingenuity if I have no idea what you're talking about? :) Is "fair bit of code" missing an "a" and is there an example of how that works?

Reply:
Well... there's _always_ a way, right? 😉 

What I've done in the past is something like this:

```typescript
// FallibleJob is a Job that is allowed to fail without compelling the worker
// process to fail.
class FallibleJob extends Job {

  constructor(name: string, image: string, event: Event) {
    super(name, image, event)
  }

  async run(): Promise<void> {
    try {
      await super.run()
    } catch {
      // Deliberately sweep any error under the rug
    }
    return Promise.resolve()
  }

}

events.on("brigade.sh/cli", "exec", async event => {
  let job0 = new FallibleJob("test-job0", "alpine", event)
  job0.primaryContainer.command = ["false"]

  let job1 = new Job("test-job1", "alpine", event)
  job1.primaryContainer.command = ["true"]
  await Job.sequence(job0, job1).run()
})
```

Rather than go off on that tangent, I think I was asking readers to take my word for it being doable.

I still don't want to go off on this tangent. Do you think it's better to just not mention that a workaround exists?

Reply:
Tentatively fixed in 73f0cf5b716271193dea9b37dab40fd2a7151dad.

I still mention the existence of workarounds, but I'm less "of coursey" about it.